[32mINFO    [0m pytest_operator.plugin:plugin.py:675 Connecting to existing model microk8s-localhost:test on unspecified cloud
[32mINFO    [0m integration.test_charm:test_charm.py:45 Deploying mysql, mysqlrouter and application
[32mINFO    [0m juju.model:model.py:2069 Deploying ch:amd64/jammy/mysql-test-app-29
[32mINFO    [0m juju.model:model.py:2069 Deploying ch:amd64/jammy/mysql-k8s-120
[32mINFO    [0m juju.model:model.py:2069 Deploying local:jammy/mysql-router-k8s-0
[32mINFO    [0m integration.test_charm:test_charm.py:76 Waiting for mysqlrouter to be in BlockedStatus
[32mINFO    [0m integration.test_charm:test_charm.py:82 Relating mysql, mysqlrouter and application
[33mWARNING [0m juju.model:model.py:1558 relate is deprecated and will be removed. Use integrate instead.
[33mWARNING [0m juju.model:model.py:1558 relate is deprecated and will be removed. Use integrate instead.
[32mINFO    [0m juju.model:model.py:2759 Waiting for model:
  mysql-router-k8s/0 [executing] maintenance: Waiting for container
[32mINFO    [0m juju.model:model.py:2759 Waiting for model:
  mysql-router-k8s/0 [idle] waiting: 
[32mINFO    [0m juju.model:model.py:2759 Waiting for model:
  mysql-router-k8s/0 [executing] active: 
[32mINFO    [0m juju.model:model.py:2759 Waiting for model:
  mysql-k8s/0 [idle] active: Primary
  mysql-k8s/1 [idle] maintenance: joining the cluster
  mysql-k8s/2 [executing] active: 
  mysql-router-k8s/0 [idle] active: 
  mysql-test-app/0 [idle] active: 
[32mINFO    [0m juju.model:model.py:2759 Waiting for model:
  mysql-router-k8s/0 [idle] active: 
[32mINFO    [0m juju.model:model.py:2759 Waiting for model:
  mysql-router-k8s (waiting for exactly 2 units, current : 1)
[32mINFO    [0m juju.model:model.py:2759 Waiting for model:
  mysql-router-k8s/0 [idle] active: 
  mysql-router-k8s/1 [executing] active: 
[32mINFO    [0m juju.model:model.py:2759 Waiting for model:
  mysql-test-app (waiting for exactly 2 units, current : 1)
[32mINFO    [0m juju.model:model.py:2759 Waiting for model:
  mysql-test-app/0 [idle] active: 
  mysql-test-app/1 [idle] active:
[32mINFO    [0m pytest_operator.plugin:plugin.py:790 Model status:

Model  Controller          Cloud/Region        Version  SLA          Timestamp
test   microk8s-localhost  microk8s/localhost  3.1.6    unsupported  01:36:09Z

App               Version                  Status  Scale  Charm             Channel      Rev  Address         Exposed  Message
mysql-k8s         8.0.35-0ubuntu0.22.04.1  active      3  mysql-k8s         8.0/edge     120  10.152.183.156  no       
mysql-router-k8s  8.0.35-0ubuntu0.22.04.1  active      2  mysql-router-k8s                 0  10.152.183.192  no       
mysql-test-app    0.0.2                    active      2  mysql-test-app    latest/edge   29  10.152.183.140  no       

Unit                 Workload  Agent  Address       Ports  Message
mysql-k8s/0*         active    idle   10.1.147.205         Primary
mysql-k8s/1          active    idle   10.1.147.207         
mysql-k8s/2          active    idle   10.1.147.206         
mysql-router-k8s/0*  active    idle   10.1.147.201         
mysql-router-k8s/1   active    idle   10.1.147.208         
mysql-test-app/0*    active    idle   10.1.147.200         
mysql-test-app/1     active    idle   10.1.147.209         

[32mINFO    [0m pytest_operator.plugin:plugin.py:796 Juju error logs:

unit-mysql-k8s-0: 01:34:12 ERROR unit.mysql-k8s/0.juju-log database-peers:3: Failed to get cluster status for cluster-a50e80ec204d0317b006a9f085ca9fc2
unit-mysql-router-k8s-0: 01:34:12 ERROR unit.mysql-router-k8s/0.juju-log Failed to connect to MySQL Server. Will retry next Juju event
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-mysql-router-k8s-0/charm/src/mysql_shell/__init__.py", line 108, in _run_code
    raise ShellDBError(**exception)
mysql_shell.ShellDBError: MySQL Error (2003): Shell.connect: Can't connect to MySQL server on 'mysql-k8s-primary.test.svc.cluster.local:3306' (111)

[32mINFO    [0m pytest_operator.plugin:plugin.py:862 Forgetting main...